src/r2mm/installing/profile_installers/GenericProfileInstaller.ts

```diff
@@ -21,6 +21,84 @@ import ZipProvider from "../../../providers/generic/zip/ZipProvider";
 
 const basePackageFiles = ["manifest.json", "readme.md", "icon.png"];
 
+
+
+abstract class PackageInstaller {
+    abstract install(mlLocation: string, modLoaderMapping: ModLoaderPackageMapping, profile: Profile): Promise<void>;
```

Comment:
Kinda annoyed that first param uses `ml` abbreviation and the second one doesn't, but I guess changing that would cause noise in other files?

Reply:
Potentially, I tried to keep changes to a minimum here so this is just the same function as before moved to a different class, didn't refactor it further than that yet.